toolz/itertoolz.py

```diff
@@ -701,27 +772,18 @@ def partition(n, seq, pad=no_pad):
     args = [iter(seq)] * n
     if pad is no_pad:
         return zip(*args)
-    else:
-        return zip_longest(*args, fillvalue=pad)
+    return zip_longest(*args, fillvalue=pad)
 
 
-def partition_all(n, seq):
-    """ Partition all elements of sequence into tuples of length at most n
-
-    The final tuple may be shorter to accommodate extra elements.
-
-    >>> list(partition_all(2, [1, 2, 3, 4]))
-    [(1, 2), (3, 4)]
+def _partition_all(n: int, seq: Sequence[_T]) -> Iterator[tuple[object, ...]]:
```

Comment:
This is a version of `partition_all` without type checking, for the actual function, its output is casted to the correct type.